hooks/taskcreator.js

```diff
@@ -0,0 +1,93 @@
+var assert      = require('assert');
+var slugid      = require('slugid');
+var taskcluster = require('taskcluster-client');
+var debug       = require('debug')('hooks:taskcreator');
+var _           = require('lodash');
+
+class TaskCreator {
+  /** Create a TaskCreator instance.
+   *
+   * options:
+   * {
+   *   credentials:   // TaskCluster credentials
+   * }
+   * */
+  constructor(options) {
+    assert(options, "options must be given");
+    assert(options.credentials instanceof Object,
+        "Expected credentials");
+    this.credentials = options.credentials;
+  }
+
+  taskForHook(hook) {
+    let task = _.cloneDeep(hook.task);
+    task.created = new Date().toJSON();
+    task.deadline = taskcluster.fromNow(hook.deadline).toJSON();
+    if (hook.expires) {
+      task.expires = taskcluster.fromNow(hook.expires).toJSON();
+    }
+    return Promise.resolve(task);
+  }
+
+  /**
+  * Fire the given hook, using the given payload (interpolating it into the task
+  * definition).  If options.taskId is set, it will be used as the taskId;
+  * otherwise a new taskId will be created.
+  */
+  async fire(hook, payload, options) {
+    options = options || {};
+    var taskId = options.taskId;
+    if (!taskId) {
+      taskId = slugid.v4();
+    }
```

Comment:
It's a matter of taste I prefer to do these with:
```js
options = _.defaults({}, options, {
  taskId: taskcluster.slugid(), // use this over slugid module
  created: new Date(),
});
```
It feels more declarative... but it's no worries.. just a random note...